docs/hello_nextflow/03_hello_containers.md

```diff
@@ -1,82 +1,442 @@
+New Docker images can be created by writing a `Dockerfile` that specifies how to build the image, but this can be a lot of work.
+Instead, we can use the Seqera Containers web service to create a container image for us.
+
+### 2.1. Navigate to Seqera Containers
+
+Navigate to [Seqera Containers web service](https://www.seqera.io/containers/) and search for the `quote` pip package.
+
+![Seqera Containers](img/seqera-containers-1.png)
+
+### 2.2. Request a container image
+
+Click on "+Add" and then "Get Container" to request a container image for the `quote` pip package.
+
+![Seqera Containers](img/seqera-containers-2.png)
+
+If this is the first time a community container has been built for this package, it may take a few minutes to complete.
+Click to copy the URI (e.g. `community.wave.seqera.io/library/pip_quote:ae07804021465ee9`) of the container image that was created for you.
+
+### 2.3. Use the container image
 
-TODO
+You can now use the container image to run the `quote` command and get a random saying from Grace Hopper.
 
-### 2.1. [step 1]
+```bash
+docker run --rm community.wave.seqera.io/library/pip_quote:ae07804021465ee9 quote "Grace Hopper"
+```
+
+Output:
+
+```console title="Output"
+Humans are allergic to change. They love to say, 'We've always done it
+this way.' I try to fight that. That's why I have a clock on my wall
+that runs counter-clockwise.
+```
 
-TODO
+### 2.4. STRETCH GOAL: Build the container image yourself
 
-### 2.2. [step 2]
+Go back to the Seqera Containers website and click on the "Build Details" button.
+There you can see the details of how our container image was built.
+You can view the conda environment file and the `Dockerfile` together make up the recipe for the container image.
+
+```conda.yml
+channels:
+- conda-forge
+- bioconda
+dependencies:
+- pip
+- pip:
+  - quote==3.0.0
+```
+
+```Dockerfile
+FROM mambaorg/micromamba:1.5.10-noble
+COPY --chown=$MAMBA_USER:$MAMBA_USER conda.yml /tmp/conda.yml
+RUN micromamba install -y -n base -f /tmp/conda.yml \
+    && micromamba install -y -n base conda-forge::procps-ng \
+    && micromamba env export --name base --explicit > environment.lock \
+    && echo ">> CONDA_LOCK_START" \
+    && cat environment.lock \
+    && echo "<< CONDA_LOCK_END" \
+    && micromamba clean -a -y
+USER root
+ENV PATH="$MAMBA_ROOT_PREFIX/bin:$PATH"
+```
+
+Copy the contents of these files into the stubs located in the `containers/build` directory, then run the following command to build the container image yourself.
+
+```bash
+docker build -t quote:latest containers/build
+```
+
+After it has finished building, you can run the container image you just built.
+
+```bash
+docker run --rm quote:latest quote "Margaret Oakley Dayhoff"
+```
+
+!!! Hint
+
+    Even if Seqera Containers doesn't manage to successfully build a container image for you, the `Dockerfile` and `conda.yml` are great starting point for a manual build.
```

Comment:
This sentence by itself is mildly concerning, it makes it sound like it might randomly work or not work. Maybe say more clearly that for cases that are not supported by Seqera, it's still possible to get some utility?